src/Select.js

```diff
@@ -372,7 +373,8 @@ export default class Select extends Component<Props, State> {
     const selectValue = cleanValue(value);
     const menuOptions = this.buildMenuOptions(props, selectValue);
 
-    this.emotion = createEmotion(props.nonce ? { nonce: props.nonce } : {});
+    this.createEmotion = memoize(createEmotion);
+    this.emotion = this.createEmotion(props.nonce ? { nonce: props.nonce } : {});
```

Comment:
`@emotion/memoize` has a simple POJO cache under the hood, it uses first argument as cache key, so u cant quite memoize based on objects - https://github.com/emotion-js/emotion/blob/fc156cf7214fda37fac3ce654569a4e17965d7e4/packages/memoize/src/index.js

Also - you have to create memoized version in the top level scope to keep cached emotion's "global", instead of caching them per `<Select/>` instance

correct way of reusing this package would look smth like this
```js
const getEmotion = memoize(nonce => createEmotion(nonce ? { nonce } : {}))

// and in constructor
this.emotion = getEmotion(props.nonce)
```